infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/LogReplicationSinkManager.java

```diff
@@ -0,0 +1,286 @@
+        this.runtime =  CorfuRuntime.fromParameters(CorfuRuntime.CorfuRuntimeParameters.builder().build())
+                .parseConfigurationString(localCorfuEndpoint).connect();
+        this.rxState = RxState.LOG_ENTRY_SYNC;
+        this.config = config;
+        init();
+    }
+
+    private void init() {
+        persistedWriterMetadata = new PersistedWriterMetadata(runtime, 0, config.getSiteID(), config.getRemoteSiteID());
+        snapshotWriter = new StreamsSnapshotWriter(runtime, config, persistedWriterMetadata);
+        logEntryWriter = new LogEntryWriter(runtime, config, persistedWriterMetadata);
+        logEntryWriter.setTimestamp(persistedWriterMetadata.getLastSrcBaseSnapshotTimestamp(),
+                persistedWriterMetadata.getLastProcessedLogTimestamp());
+        readConfig();
+        sinkBufferManager = new SinkBufferManager(LOG_ENTRY_MESSAGE, ackCycleTime, ackCycleCnt, bufferSize,
+                persistedWriterMetadata.getLastProcessedLogTimestamp(), this);
+    }
+
+    private void readConfig() {
+        File configFile = new File(config_file);
+        try {
+            FileReader reader = new FileReader(configFile);
+            Properties props = new Properties();
+            props.load(reader);
+
+            bufferSize = Integer.parseInt(props.getProperty("log_writer_queue_size"));
+            ackCycleCnt = Integer.parseInt(props.getProperty("log_writer_ack_cycle_count"));
+            ackCycleTime = Integer.parseInt(props.getProperty("log_writer_ack_cycle_time"));
+            reader.close();
+        } catch (FileNotFoundException e) {
+            log.warn("Config file {} does not exist.  Using default configs", config_file);
+            bufferSize = DefaultSiteConfig.getLogWriterQueueSize();
+            ackCycleCnt = DefaultSiteConfig.getLogWriterAckCycleCount();
+            ackCycleTime = DefaultSiteConfig.getLogWriterAckCycleTime();
+        } catch (IOException e) {
+            log.error("IO Exception when reading config file", e);
+        }
+        logEntryWriter.setMaxMsgQueSize(bufferSize);
+        log.info("Log writer config queue size {} ackCycleCnt {} ackCycleTime {}",
+                bufferSize, ackCycleCnt, ackCycleTime);
+    }
+
+    @Override
+    public LogReplicationEntry receive(LogReplicationEntry message) {
+
+        //If the roletype is active or it is not the leader, skip processing the message and ack the timestamp.
+        //This will  trigger the active site to redo the query leadership.
+        if (active || !leader) {
+            LogReplicationEntryMetadata metadata = sinkBufferManager.makeAckMessage(message);
+            return new LogReplicationEntry(metadata, new byte[0]);
+        }
+
+        log.info("Sink manager received {} while in {}", message.getMetadata().getMessageMetadataType(), rxState);
+
+        siteConfigID = Math.max(siteConfigID, message.getMetadata().getSiteConfigID());
```

Comment:
what does this mean? why keep the highest? Isn't the Site Config ID what it is, not changing...